source/slang/slang-ir-autodiff.cpp

```diff
@@ -126,6 +126,12 @@ static IRInst* _getDiffTypeWitnessFromPairType(
 
 bool isNoDiffType(IRType* paramType)
 {
+    if (auto attrType = as<IRAttributedType>(paramType))
```

Comment:
It seems to me that we need to fuse the two loops below into one, instead of adding another check here.